### DISCUSSION TITLE
Required EVP_PKEY_get_raw_public_key() Not Found in OpenSSL Library

### DISCUSSION THREAD

**Discussion by kinghelong** [SoftEtherVPN/SoftEtherVPN#2012] [General]
Hello,

I am trying to compile SoftEther VPN on my Windows 11 system using CMake. However, I encountered an issue related to the OpenSSL library. Below is the error message I received:
CMake Error at src/Mayaqua/CMakeLists.txt:34 (message):
  Required EVP_PKEY_get_raw_public_key() not found in OpenSSL library!
Environment:
Operating System: Windows 11
CMake Version: 3.3
OpenSSL Version: 3.3 (also tried with other versions)
Additional Information:
The function EVP_PKEY_get_raw_public_key() is available in OpenSSL 1.1.1+ but it seems the build process is not detecting it correctly.
I have successfully compiled OpenSSL separately and verified its installation.
Request:
Could you please provide guidance on how to resolve this issue? Is there a specific version of OpenSSL that I should be using, or are there additional configuration steps required?

Thank you for your assistance!

Best regards,

**Comment by chipitsine** [comment]
it might be interesting to try a build in Github Actions.
it should be easy, just enable them in your own fork.

I think we might be hitting vcpkg cache in our Github Actions, not sure which OpenSSL is used there

**Comment by chipitsine** [comment]
do you use VIsual Studio ?

I've updated my local vcpkg, and I see no problem building SE VPN against OpenSSL 3.3.1

<img width="900" alt="image" src="https://github.com/SoftEtherVPN/SoftEtherVPN/assets/2217296/cd66d1d2-5f0f-4966-b8a5-c5dc5d1e767e">

**Comment by kinghelong** [comment]
Thank you for your response and assistance. After further consideration and attempts, I've decided to abandon this approach. The complexity of the project is beyond my current capabilities, and I'll wait for potential updates or solutions in the future. Thanks again for your help.

**Comment by chipitsine** [comment]
I agree that vcpkg may be non trivial, but we intended that for regular developer it shouldn't be harder than just open project in Visual Studio ((
